src/main/java/bio/terra/cli/command/resource/ListTree.java

```diff
@@ -0,0 +1,89 @@
+package bio.terra.cli.command.resource;
+
+import bio.terra.cli.businessobject.Context;
+import bio.terra.cli.businessobject.Resource;
+import bio.terra.cli.command.shared.BaseCommand;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.serialization.userfacing.UFResource;
+import bio.terra.workspace.model.Folder;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import picocli.CommandLine;
+
+/** This class corresponds to the third-level "terra resource list-tree" command. */
+@CommandLine.Command(
+    name = "list-tree",
+    description = "List all resources and folders in tree view.")
+public class ListTree extends BaseCommand {
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+
+  private final HashMap<UUID, ArrayList<UUID>> edges = new HashMap<>();
```

Comment:
all these fields should be private static final. and change the variable name to all cap